agent/pkg/tunnel/protocol/tcp/tcpproxy.go

```diff
@@ -0,0 +1,163 @@
+package tcp
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kubeedge/edgemesh/agent/pkg/tunnel/controller"
+	tcp_pb "github.com/kubeedge/edgemesh/agent/pkg/tunnel/protocol/tcp/pb"
+	"github.com/kubeedge/edgemesh/common/constants"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/libp2p/go-msgio/protoio"
+	"k8s.io/klog/v2"
+)
+
+var TCPProxyProtocol protocol.ID = "/libp2p/tcpproxy/1.0.0"
+
+type TCPProxyService struct {
+	host host.Host
+}
+
+func NewTCPProxyService(h host.Host) *TCPProxyService {
+	return &TCPProxyService{
+		host: h,
+	}
+}
+
+func (tp *TCPProxyService) ProxyStreamHandler(s network.Stream) {
+	// todo use peerID to get nodeName
+	klog.Infof("Get a new stream from %s", s.Conn().RemotePeer().String())
+	streamWriter := protoio.NewDelimitedWriter(s)
+	streamReader := protoio.NewDelimitedReader(s, 4*1024)
```

Comment:
`4*1024` needs to be defined as a constant to show its meaning

Reply:
yes, i will modify